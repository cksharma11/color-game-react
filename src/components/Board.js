import React from "react";
import Box from "./Box";
import getRandom from "../util";

export default class Board extends React.Component {
  render() {
    return (
      <div className="board">
        <div
          id="current-color"
          className="text"
          style={{ color: this.props.colors[getRandom()] }}
        >
          {this.props.currentColor.toUpperCase()}
        </div>
        <div className="row">
          <Box color={this.props.colors[0]} onClick={this.props.onClick} />
          <Box color={this.props.colors[1]} onClick={this.props.onClick} />
        </div>
        <div className="row">
          <Box color={this.props.colors[2]} onClick={this.props.onClick} />
          <Box color={this.props.colors[3]} onClick={this.props.onClick} />
        </div>
        <div
          id="score"
          className="text"
          style={{ color: this.props.score >= 0 ? "green" : "red" }}
        >
          Score : {this.props.score}
        </div>
      </div>
    );
  }
}
