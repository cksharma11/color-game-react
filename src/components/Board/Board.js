import React, { useContext } from "react";
import Box from "../Box/Box";
import getRandom from "../../util";
import GameContext from "../../contexts/GameContext";

const Board = () => {
  const gameContext = useContext(GameContext.Context);
  const { gameColors = {}, score, currentColor = "", gameScoreHandler } = gameContext;

  return (
    <div className="board">
      <div id="current-color" className="text" style={{ color: gameColors[getRandom()] }}>
        {currentColor.toUpperCase()}
      </div>
      <div className="row">
        <Box color={gameColors[0]} onClick={gameScoreHandler} />
        <Box color={gameColors[1]} onClick={gameScoreHandler} />
      </div>
      <div className="row">
        <Box color={gameColors[2]} onClick={gameScoreHandler} />
        <Box color={gameColors[3]} onClick={gameScoreHandler} />
      </div>
      <div id="score" className="text" style={{ color: score >= 0 ? "green" : "red" }}>
        Score : {score}
      </div>
    </div>
  );
};

export default Board;
