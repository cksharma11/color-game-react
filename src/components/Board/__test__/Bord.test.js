import { shallow, mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";
import Board from "../Board";

configure({ adapter: new Adapter() });

describe("Board", () => {
  const colors = ["red", "blue", "green", "yellow"];

  it("should render properly", () => {
    const props = {
      colors: colors,
      currentColor: "red",
      score: { score: 0 },
      onClick: () => {}
    };

    const wrapper = shallow(<Board {...props} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
