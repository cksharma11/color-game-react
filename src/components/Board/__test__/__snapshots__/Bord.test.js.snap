// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
"<div className=\\"board\\">
  <div id=\\"current-color\\" className=\\"text\\" style={{...}}>
    RED
  </div>
  <div className=\\"row\\">
    <Box color=\\"red\\" onClick={[Function: onClick]} />
    <Box color=\\"blue\\" onClick={[Function: onClick]} />
  </div>
  <div className=\\"row\\">
    <Box color=\\"green\\" onClick={[Function: onClick]} />
    <Box color=\\"yellow\\" onClick={[Function: onClick]} />
  </div>
  <div id=\\"score\\" className=\\"text\\" style={{...}}>
    Score : 
    {{ score: 0 }}
  </div>
</div>"
`;

exports[`Board should render properly 1`] = `
"<div className=\\"board\\">
  <div id=\\"current-color\\" className=\\"text\\" style={{...}}>
    RED
  </div>
  <div className=\\"row\\">
    <Box color=\\"red\\" onClick={[Function: onClick]} />
    <Box color=\\"blue\\" onClick={[Function: onClick]} />
  </div>
  <div className=\\"row\\">
    <Box color=\\"green\\" onClick={[Function: onClick]} />
    <Box color=\\"yellow\\" onClick={[Function: onClick]} />
  </div>
  <div id=\\"score\\" className=\\"text\\" style={{...}}>
    Score : 
    {{ score: 0 }}
  </div>
</div>"
`;
